src/cpp/session/modules/SessionFonts.cpp

```diff
@@ -0,0 +1,338 @@
+/*
+ * SessionFonts.cpp
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include "SessionFonts.hpp"
+
+#include <core/http/Request.hpp>
+#include <core/http/Response.hpp>
+#include <core/Exec.hpp>
+#include <core/text/TemplateFilter.hpp>
+
+#include <core/system/Xdg.hpp>
+
+#include <shared_core/FilePath.hpp>
+#include <shared_core/SafeConvert.hpp>
+
+#include <session/SessionModuleContext.hpp>
+
+#include <array>
+
+using namespace rstudio::core;
+
+#define kFontFolder    "fonts"
+#define kFontsLocation "fonts/"
+#define kFontFiles     kFontsLocation "files/"
+#define kFontCss       kFontsLocation "css/"
+
+namespace rstudio {
+namespace session {
+namespace modules {
+namespace fonts {
+
+namespace {
+
+// File extensions that we recognize as font files
+std::array<std::string, 5> s_fontExtensions = {
+   ".eot",
+   ".ttf",
+   ".otf",
+   ".woff",
+   ".woff2"
+};
+
+// Indicates whether the file is a font file (i.e. has a known font file extension)
+bool isFontFile(const FilePath& file)
+{
+   // Test the file extension against known font formats
+   std::string ext = file.getExtensionLowerCase();
+   for (size_t i = 0; i < s_fontExtensions.size(); i++)
```

Comment:
Could use: `for (const auto& fontExtension : s_fontExtensions)` here and `if (ext == fontExtension)` below.